src/components/freelancerJobs/freelancerPage.tsx

```diff
@@ -0,0 +1,213 @@
+import { FC, useEffect, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Link } from 'react-router-dom';
+import Typography from '@material-ui/core/Typography';
+import Slider from '@material-ui/core/Slider';
+import { useGetJobPostsQuery } from 'service/httpService';
+import {
+	TitleStyled,
+	DescriptionDataStyled,
+	ColumnSmall,
+	ColumnBig,
+	FilterLabel,
+	SkillsLabel,
+	SkillsButtonsBlock,
+	ClearBtn,
+	CategoryDiv,
+	SearchInput,
+	CustomSelect,
+	P,
+	Li,
+	CheckLabel,
+	Span,
+	Column,
+	Label,
+} from 'components/freelancerJobs/freelancerPage.styles';
+import {
+	Category,
+	checkList,
+	IPost,
+	ISkill,
+	selection,
+	skillsMock,
+} from 'components/freelancerJobs/changes';
+import { useAppSelector } from 'redux/hooks';
+import { RootState } from 'redux/store';
+
+const intitialState = {
+	value: '',
+	label: '',
+};
+
+const FreelancerPage: FC = () => {
+	const { t } = useTranslation();
+	const { user } = useAppSelector<RootState>(state => state);
+	const { data: posts } = useGetJobPostsQuery(user.id);
+
+	const [skillsOptions, setSkillsOptions] = useState<ISkill[]>(skillsMock);
+	const filteredSkills = skillsOptions.filter(s => s.value);
+	const skills = filteredSkills.map(skill => {
+		return skill.name;
+	});
+
+	const onSkillsChange = (index: number) => {
+		setSkillsOptions(prevState => {
+			return prevState.map((e, i) => {
+				if (index === i) {
+					return { ...e, value: !e.value };
+				}
+				return e;
+			});
+		});
+	};
+
+	const optionButtons = useMemo(() => {
+		return skillsOptions.map((e, i) => (
+			<Label key={e.name} className={`btn btn-${e.value ? 'primary' : 'light'}`}>
+				<input
+					type="checkbox"
+					checked={e.value}
+					autoComplete="off"
+					onChange={() => onSkillsChange(i)}
+				/>
+				{e.name}
+			</Label>
+		));
+	}, [skillsOptions]);
+
+	const [search, setSearch] = useState<string>('');
```

Comment:
it was not fixed. I explaine the order
hooks, handles, renders
now its mess

Reply:
changed